pkg/sync/kubernetes/kubernetes_sync_test.go

```diff
@@ -556,6 +559,281 @@ func TestSync_watcher(t *testing.T) {
 	}
 }
 
+func TestInit(t *testing.T) {
+	t.Run("expect error with wrong URI format", func(t *testing.T) {
+		k := Sync{URI: ""}
+		e := k.Init(context.TODO())
+		if e == nil {
+			t.Errorf("Expected error but got none")
+		}
+		if k.IsReady() {
+			t.Errorf("Expected NOT to be ready")
+		}
+	})
+	t.Run("expect informer registration", func(t *testing.T) {
+		const name = "myFF"
+		const ns = "myNS"
+		scheme := runtime.NewScheme()
+		ff := &unstructured.Unstructured{}
+		ff.SetUnstructuredContent(getCFG(name, ns))
+		fakeClient := fake.NewSimpleDynamicClient(scheme, ff)
+		k := Sync{
+			URI:           fmt.Sprintf("%s/%s", ns, name),
+			DynamicClient: fakeClient,
+			namespace:     ns,
+		}
+		e := k.Init(context.TODO())
+		if e != nil {
+			t.Errorf("Unexpected error: %v", e)
+		}
+		if k.informer == nil {
+			t.Errorf("Informer not initialized")
+		}
+		if k.IsReady() {
+			t.Errorf("The Sync should not be ready")
+		}
+	})
+}
+
+func TestSync(t *testing.T) {
+	const name = "myFF"
+	const ns = "myNS"
+	s := runtime.NewScheme()
+	ff := &unstructured.Unstructured{}
+	ff.SetUnstructuredContent(getCFG(name, ns))
+	fakeDynamicClient := fake.NewSimpleDynamicClient(s, ff)
+	validFFCfg := &v1alpha1.FeatureFlagConfiguration{
+		TypeMeta: Metadata,
+		ObjectMeta: v1.ObjectMeta{
+			Name:      name,
+			Namespace: ns,
+		},
+	}
+	fakeReadClient := newFakeReadClient(validFFCfg)
+	l, _ := logger.NewZapLogger(zapcore.DebugLevel, "")
+
+	t.Run("Happy Path", func(t *testing.T) {
```

Comment:
Could [table driven tests](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) be used here to reduce duplication/improve readability?

Reply:
I think we can do that 👍